Arduino boards integration examples

kaloyanam opened the pull request.

desislava-marinova commented:
Hi @kaloyanam ,
I have some comments on this PR. Could you please
- provide readme what this code is for, how to use it
- It will be good to use (where possible) vendor-less library names, e.g. BoschIoTAgent to IoTAgent, etc.
- some of the files are missing the header

kaloyanam commented:
Hi @desislava-marinova,
 
> * provide readme what this code is for, how to use it

Readme file is provided

> * It will be good to use (where possible) vendor-less library names, e.g. BoschIoTAgent to IoTAgent, etc.

Removed vendor names from library names. It was initially called that way because it was tested with the Bosch IoT Suite, but I agree that the library name should not contain vendors

> * some of the files are missing the header

Added missing headers
